tsdb/querier.go

```diff
@@ -413,7 +414,12 @@ func postingsForMatcher(ix IndexReader, m *labels.Matcher) (index.Postings, erro
 	if m.Type == labels.MatchRegexp {
 		setMatches := findSetMatches(m.Value)
 		if len(setMatches) > 0 {
-			return postingsForSetMatcher(ix, m.Name, setMatches)
+			sort.Strings(setMatches)
+			it, err := ix.Postings(m.Name, setMatches...)
+			if err != nil {
+				return nil, err
+			}
+			return it, nil
```

Comment:
Why cant this be just `return ix.Postings(m.Name, setMatches...)`?

Reply:
Done